test/versioned/opensearch/opensearch.test.js

```diff
@@ -0,0 +1,475 @@
+/*
+ * Copyright 2024 New Relic Corporation. All rights reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+'use strict'
+const test = require('node:test')
+const assert = require('node:assert')
+const helper = require('../../lib/agent_helper')
+const params = require('../../lib/params')
+const urltils = require('../../../lib/util/urltils')
+const crypto = require('crypto')
+const DB_INDEX = `test-${randomString()}`
+const DB_INDEX_2 = `test2-${randomString()}`
+const SEARCHTERM_1 = randomString()
+
+function randomString() {
+  return crypto.randomBytes(5).toString('hex')
+}
+
+function setRequestBody(body) {
+  return { body }
+}
+function setBulkBody(body) {
+  return {
+    refresh: true,
+    body
+  }
+}
+function setMsearch(body) {
+  return { body }
+}
+
+test('opensearch instrumentation', async (t) => {
+  t.beforeEach(async (ctx) => {
+    const agent = helper.instrumentMockedAgent()
+
+    const METRIC_HOST_NAME = urltils.isLocalhost(params.elastic_host)
```

Comment:
Are these still `elastic_host` and `elastic_port` in OpenSearch? I'm poking through their docs and haven't found confirmation yet.

Reply:
Ah, wait--I was reading that as params from OS, rather than ours.

Reply:
I could try to add a new container that runs opensearch instead of elasticsearch. I won't have time to do that today though, I can explore after holiday

Reply:
actually it wasn't too bad, i'll push the update